gen3utils/gitops/gitops_validator.py

```diff
@@ -0,0 +1,385 @@
+        props,
+        errors,
+    )
+    return errors
+
+
+def _validate_studyViewerConfig_helper(viewer, type_prop_map, errors):
+    datatype = viewer.get("dataType")
+    props = type_prop_map.get(datatype, [])
+    if not props:
+        errors.append(
+            FieldError(
+                "Invalid field {} in studyViewerConfig.dataType".format(datatype)
+            )
+        )
+
+    listitemconfig = viewer.get("listItemConfig")
+    errors = _validate_itemConfig(listitemconfig, props, errors)
+    if viewer.get("singleItemConfig"):
+        errors = _validate_itemConfig(viewer.get("singleItemConfig"), props, errors)
+
+    for dtype, props in type_prop_map.items():
+        if viewer["rowAccessor"] not in props:
+            errors.append(
+                FieldError(
+                    "rowAccessor {} not found in index with type {}".format(
+                        viewer["rowAccessor"], dtype
+                    )
+                )
+            )
+
+    return errors
+
+
+def validate_studyViewerConfig(studyviewer, type_prop_map, errors):
+    for viewer in studyviewer:
+        errors = _validate_studyViewerConfig_helper(viewer, type_prop_map, errors)
+    return errors
+
+
+def _validate_explorerConfig_helper(explorer_config, type_prop_map, errors):
+
+    datatype = explorer_config["guppyConfig"]["dataType"]
+    props = type_prop_map.get(datatype, [])
+    if not props:
+        errors.append(
+            FieldError(
+                "Invalid field {} in explorerConfig.guppyConfig.dataType".format(
+                    datatype
+                )
+            )
+        )
+
+    tabs = explorer_config["filters"]["tabs"]
+    for tab in tabs:
+        errors = check_field_value(
+            "explorerConfig.filters.tabs.fields", tab.get("fields", []), props, errors
+        )
+
+    table = explorer_config["table"]
+    if table["enabled"]:
+        errors = check_field_value(
+            "explorerConfig.table.fields", table.get("fields", []), props, errors
+        )
+
+    manifest_map = explorer_config.get("manifestMapping")
+    if manifest_map and manifest_map.get("resourceIndexType"):
+        resource_props = type_prop_map.get(manifest_map.get("resourceIndexType"))
+        if not resource_props:
+            errors.append(
+                FieldError(
+                    "Invalid field {} in manifestMapping.resourceIndexType".format(
+                        manifest_map.get("resourceIndexType")
+                    )
+                )
+            )
+        elif manifest_map.get("resourceIdField") not in resource_props:
+            errors.append(
+                FieldError(
+                    "Invalid field {} in manifestMapping.resourceIdField".format(
+                        manifest_map.get("resourceIdField")
+                    )
+                )
+            )
+        # TODO
+        # Also consider fields referenceIdFieldInResourceIndex and referenceIdFieldInDataIndex
+
+    return errors
```

Comment:
i think you don't need to return `errors` since you're updating in place the list given as parameter